scripts/github_pr/github_pr.rb

```diff
@@ -0,0 +1,405 @@
+    description ||= RESULT_MESSAGES[status]
+    description.to_s
+  end
+
+  def get_full_sha_status(sha)
+    status = GithubAPI.new.client.status(org_repo, sha)
+    status.statuses.select{ |s| s.context == context }.first rescue {}
+  end
+
+  def get_sha_status(sha)
+    get_full_sha_status(sha).state rescue ''
+  end
+
+  def get_pr(pr)
+    GithubAPI.new.client.pull_request(org_repo, pr)
+  end
+
+  def get_pr_info(pr)
+    get_pr(pr).to_attrs
+  end
+
+  def get_pr_latest_sha(pr)
+    get_pr(pr).head.sha rescue ''
+  end
+
+  def is_latest_sha?(pr, sha)
+    get_pr_latest_sha(pr) == sha
+  end
+
+  # state = ["open"|"closed"]
+  def get_all_pull_requests(state)
+     GithubAPI.new.client.pull_requests(org_repo, :state => state)
+  end
+end
+
+class GithubPRWorker
+  def initialize(base_parameters = {}, config = {})
+    @base_parameters = base_parameters
+    @config = config
+  end
+
+  def ghc_ctx(org, repo, ctx)
+    {
+      :org_repo => org + "/" + repo,
+      :organization => org,
+      :repository => repo,
+      :context => ctx,
+      :config_base_path => File.dirname(@base_parameters[:config]),
+    }
+  end
+
+  def process_list
+    @config["pr_processing"] || []
+  end
+
+  def status_filter_config(config)
+    # the config for the Status filter might be overridden via command line parameters
+    if (@base_parameters.has_key?(:mode) && @base_parameters[:mode].to_s.size > 0) then
+      config["status"] = @base_parameters[:mode]
+    end
+    config
+  end
+
+  def debug?
+    @base_parameters.has_key?(:debugfilterchain) && \
+      @base_parameters[:debugfilterchain] == true
+  end
+
+  def debug_filterchain(pulls, filter)
+    return unless debug?
+    return unless filter
+    puts "=" * 25
+    puts filter.inspect
+    print_prs(pulls)
+    puts "-" * 25
+  end
+
+  def run_actions(pulls, actions)
+    return unless actions.is_a?(Array)
+    actions.each do |a|
+      a.run(pulls)
+    end
+  end
+
+  def get_repos(item)
+    if (item["config"].has_key?("repositories") && item["config"]["repositories"].size > 0) then
+      item["config"]["repositories"]
+    elsif (item["config"].has_key?("repository_filter") && item["config"]["repository_filter"].size > 0) then
+      c = GithubAPI.new.client
+      c.repositories(item["config"]["organization"]).collect do |r|
+        r.name if item["config"]["repository_filter"].find do |rf|
+          r.name =~ rf
+        end
+      end.compact
+    end
+  end
+
+  def run_filterchain(filterchain, mode, white)
+    filterchain.each do |h|
+      white, black = h[:filter].filter(white)
+      debug_filterchain(white, h[:filter])
+      if (mode == :process) then
+         run_actions(black, h[:blacklist_handler])
+         run_actions(white, h[:whitelist_handler])
+      end
+      debug_filterchain(black, h[:blacklist_handler])
+      debug_filterchain(white, h[:whitelist_handler])
+    end
+    return white
+  end
+
+  def filter_prs(mode = :get, state = :open)
```

Comment:
Abbreviations should be avoided for method names, so please us `filter_pull_requests`.

Reply:
Ok, I will stick to the names used on other places in the tool: `pull` resp. `pulls`